ansible/files/bootstrap/consul/post/test/vault-integration.yml

```diff
@@ -1,78 +1,87 @@
+### TESTS policies
+#######################
+# ACL
+- name: "{{ service }} - Debug tokens"
+  debug:
+    msg: "admin token {{ consul.vault.acl.admin_token }}\nproducer token {{consul.vault.acl.producer_token}}\nconsumer token {{consul.vault.acl.consumer_token}}"
+
+- name: "{{ service }} - list acl - admin token - allow"
+  shell: consul acl token list
+  environment:
+    CONSUL_HTTP_TOKEN: "{{ consul.vault.acl.admin_token }}"
+
+- name: "{{ service }} - list acl - producer token - deny"
+  shell: consul acl token list
+  register: consul_acl_list_try1
+  environment:
+    CONSUL_HTTP_TOKEN: "{{ consul.vault.acl.producer_token }}"
+  failed_when: consul_acl_list_try1.rc == 0
+
+- name: "{{ service }} - list acl - consumer token - deny"
+  shell: consul acl token list
+  register: consul_acl_list_try2
+  environment:
+    CONSUL_HTTP_TOKEN: "{{ consul.vault.acl.consumer_token }}"
+  failed_when: consul_acl_list_try2.rc == 0
+
+# K/V
+## 1 ADMIN
+- name: "{{ service }} - write K/V to root '/' - admin token - allow"
   uri:
     # curl -v -X PUT --header "X-Consul-Token: <token>" -d 'my-admin-value' http://127.0.0.1:8500/v1/kv/my-admin-key
     url:  http://127.0.0.1:8500/v1/kv/my-admin-key
     method: PUT
-    body: "my-admin-value"
-    status_code: 403
-    return_content: yes
-    headers:
-      X-Consul-Token: "{{ consul.vault.acl.admin_team_token }}"
-
-- name: "{{ service }} - post/test/vault-integration - admin token - write key-value to /team/admin is allowed"
-  uri:
-    # curl -v -X PUT --header "X-Consul-Token: <token>" -d 'my-admin-value' http://127.0.0.1:8500/v1/kv/team/admin/my-admin-key
-    url:  http://127.0.0.1:8500/v1/kv/team/admin/my-admin-key
-    method: PUT
-    body: "my-admin-value"
     status_code: 200
-    return_content: yes
+    body: "my-admin-value"
     headers:
-      X-Consul-Token: "{{ consul.vault.acl.admin_team_token }}"
+      X-Consul-Token: "{{ consul.vault.acl.admin_token }}"
 
-- name: "{{ service }} - post/test/vault-integration - admin token - list all the consul ACL policies is allowed"
+## 1 PRODUCER
+- name: "{{ service }} - write K/V to root '/' - producer token - deny"
   uri:
-    # curl --header "X-Consul-Token: <token>" http://127.0.0.1:8500/v1/acl/policies
-    url: http://127.0.0.1:8500/v1/acl/policies
-    method: GET
-    headers:
-      X-Consul-Token: "{{ consul.vault.acl.admin_team_token }}"
-    status_code: 200
-    return_content: yes
-    body_format: json
-  register: result_policies
-
-# Test dev.team.policy
-- name: "{{ service }} - post/test/vault-integration - dev token - write key-value to root '/' is forbidden"
-  uri:
-    # curl -v -X PUT --header "X-Consul-Token: <token>" -d 'my-dev-value' http://127.0.0.1:8500/v1/kv/my-dev-key
-    url:  http://127.0.0.1:8500/v1/kv/my-dev-key
+    # curl -v -X PUT --header "X-Consul-Token: <token>" -d 'my-producer-value' http://127.0.0.1:8500/v1/kv/will-fail-key
+    url:  http://127.0.0.1:8500/v1/kv/will-fail-key
```

Comment:
Is the consul address (127.0.0.1:8500) set as an environment variable? 
Would be easier if we could use something like `{{ consul_http_addr }}`

Reply:
Need to double-check interpolation when URL uses